bin/build_and_test_pod

```diff
@@ -24,4 +24,4 @@ echo "--- :test_tube: Building and Running Tests"
 # https://github.com/Automattic/buildkite-ci/issues/7
 xcrun simctl list >> /dev/null
 
-bundle exec fastlane $LANE_NAME
+bundle exec fastlane "$LANE_NAME"
```

Comment:
We should extra-check that none of our existing pipelines rely on that glitch btw. Like if one calls `build_and_test_pod "test param1:value1"` in an attempt to pass custom params to the test lane (which seems like a reasonable need to have), this fix will now break that behavior. Imho that's still a good thing as the proper solution instead would be to make `build_and_test_pod` accept optional parameters (in addition to `$1` aka `$LANE_NAME`) explicitly, to make it clear that this is supported, instead of making caller rely on the bug and implementation details.

Reply:
Actually there's another use case where this would break, which is when passing an explicit platform to run either ios or macos tests.

This is actually what's used in https://github.com/Automattic/MobilePay-Apple/blob/develop/.buildkite/pipeline.yml#L15 – which this fix would break.

We should thus fix this in another way instead, making the whole `build_and_test_pod` script allow arbitrary parameters other than `$LANE_NAME`, but instead likely use `$@`, and make it clear and explicit that _every_ parameter (`%@`) passed to that script will be passed along to `fastlane` unchanged, allowing you to pass arbitrary platform name before lane name and arbitrary parameters (like `foo:bar` but also flags like `--verbose`) to fastlane if needed.

Reply:
It should be possible to step this out – apply this fix and ship a new version of the plugin, then update that repo. Assuming the projects are well-behaved, they should always be pulling based on a tag.

I like the idea of the "splat operator" approach where we could "pass through" args.

Reply:
Reverted in d08a95f (+ added comment).

I'll make a separate PR on top of this to include that fix separately in preparation for a future breaking version bump.

Reply:
Separate PR to prepare the major version bump now ready here: https://github.com/Automattic/bash-cache-buildkite-plugin/pull/13/files